src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -146,11 +146,21 @@ def __init__(
 
     @property
     def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
+        from snowflake.snowpark._internal.analyzer.analyzer import ARRAY_BIND_THRESHOLD
+
         # select $1, ..., $m FROM VALUES (r11, r12, ..., r1m), (rn1, ...., rnm)
-        # TODO: use ARRAY_BIND_THRESHOLD
+        literal_complexity = len(self.data) * len(self.output)
+        if literal_complexity >= ARRAY_BIND_THRESHOLD:
+            # When the number of literals exceeds the threshold, we generate 3 queries:
```

Comment:
It's not clear to me how this comment relate to the return statement below it.

Reply:
I updated the comment here to make things more clear.